.github/workflows/static-analysis-js.yml

```diff
@@ -39,16 +44,16 @@ jobs:
           node-version: ${{ inputs.NODE_VERSION }}
           registry-url: ${{ inputs.NPM_REGISTRY_DOMAIN }}
 
-      - name: Cache npm dependencies
+      - name: Cache npm/Yarn dependencies
         uses: actions/cache@v3
         with:
-          path: ~/.npm
-          key: ${{ runner.os }}-node-${{ hashFiles('**/package-lock.json') }}
-          restore-keys: |
-            ${{ runner.os }}-node-
+          path: ~/.deps_cache
```

Comment:
IMO, it would be better to read the cache directory from the selected package manager instead of hardcoding it to `~/.deps_cache`. This would also allow us to remove the `${{ inputs.PACKAGE_MANAGER }}` part from the `key` config in the next line because npm and Yarn use different folders.

But please see the next comment on the challenges to get/set the cache directory for Yarn.

Reply:
Hmm, yes it's a good suggestion and it follows author guidelines - https://github.com/actions/cache/blob/main/examples.md#get-npm-cache-directory-using-same-shell

But if we are using `@actions/setup-node` maybe we can just use build-in functionality https://github.com/actions/setup-node/blob/main/docs/advanced-usage.md#caching-packages-data. With this approach, we can remove the dependency step by just
```
cache: ${{ inputs.PACKAGE_MANAGER }}
```

The additional benefit it should support Yarn 1 and 2 out-of-the-box - https://github.com/actions/setup-node/blob/main/src/cache-utils.ts#L68-L77

Reply:
Yes, utilizing existing tools for this would be the best approach. Please apply any changes to all workflows that install npm dependencies. Thanks!